vme/src/unit_data.h

```diff
@@ -112,21 +114,31 @@ class unit_data : public basedestruct
     file_index_type *getFileIndex();
     const file_index_type *getFileIndex() const;
     void setFileIndex(file_index_type *value);
+
     [[nodiscard]] const char *getFileIndexZoneName() const
     {
         if (m_fi)
         {
             return m_fi->getZone()->getName();
         }
-        return "NO-ZONE";
+        return NO_ZONE;
     }
+
     [[nodiscard]] const char *getFileIndexName() const
     {
         if (m_fi)
         {
             return m_fi->getName();
         }
-        return "NO-NAME";
+        return NO_NAME;
+    }
+
+    [[nodiscard]] std::string getFileIndexSymName() const
```

Comment:
Question. How is it that the string frees itself? It returns an object. But the receiver doesn't delete it... So how does it go away without a GC? :-)

Reply:
It becomes a "temporary object" and because nothing holds a handle to it, it is destroyed by the end of the statement. But say something did
`auto &foobar=u->getFileIndexSymName();`
It would exist unit foobar goes out of scope. They are also called r-values and pr-values - and it can get complicated but basically the compiler has an object with no name/reference to it., so destroys it when its finished. 

Luckily there is no GC in C++ ;-)